src/shape.rs

```diff
@@ -211,8 +211,10 @@ impl<'a, 'b, B: Brush> partition::Selector for FontSelector<'a, 'b, B> {
     type SelectedFont = SelectedFont;
 
     fn select_font(&mut self, cluster: &mut CharCluster) -> Option<Self::SelectedFont> {
+        use fontique::QueryFamily;
         let style_index = cluster.user_data() as u16;
-        if style_index != self.style_index {
+        let is_emoji = cluster.info().is_emoji();
+        if style_index != self.style_index || is_emoji {
             self.style_index = style_index;
```

Comment:
Do we also need a `self.is_emoji`? I can see this potentially failing to switch back to a regular font after an emoji...

Reply:
Good catch. Checking for `self.font_id.is_none()` should be sufficient. Ultimately I’d like to rework this logic.

Reply:
Added this additional check which will force update of the current font stack following an emoji cluster.